### ISSUE TITLE
Choose the right order of eigenvector from the PCA axes to create the coordinate system

### ISSUE THREAD

**Issue by ommmid** [ommmid/supertoroid_paper#2]
The idea is to come up with a method that chooses the right coordinate axes from the PCA axes. Currently, distance between the point cloud and the fitted supertoroid is calculated three times; one for each possible candidate which increase the time complexity

**Comment by ommmid** [comment]
first question is are we considering only a subgroup of all possible shapes, for example only torus? the criteria that chooses the z axis to be in the middle for a torus might not work for other shapes.

**Comment by albarebecucha** [comment]
Yes, it is supertoroid fitting, so we want the z axis in the "hole". We are not optimizing other shapes but whatever methodology we use, we need to consider that if there is no hole (super ellipsoid), the results are also valid. It is a bit tricky because in that case, we want the eigenvector of the smallest eigenvalue to coincide with the z direction.
